odoo/addons/base/models/assetsbundle.py

```diff
@@ -81,11 +86,39 @@ def __init__(self, name, files, external_assets=(), env=None, css=True, js=True,
                 'inline': f['content'],
                 'last_modified': None if self.is_debug_assets else f.get('last_modified'),
             }
-            if css:
+            if css and not binary and extension in STYLE_EXTENSIONS:
+                if css_urls is None and self.binaries:
+                    def relative_path(file, folder):
+                        # pathlib.Path.relative_to does not support '..'
+                        # => try to replicate the behavior of os.path.relpath
+                        file_parts = Path(file).resolve().parts
+                        folder_parts = Path(folder).resolve().parts
+                        common_index = 0
+                        while common_index < min(len(file_parts), len(folder_parts)) and file_parts[common_index] == folder_parts[common_index]:
+                            common_index += 1
+                        up = ['..'] * (len(folder_parts) - common_index)
+                        down = file_parts[common_index:]
+                        return str(Path(*up, *down))
```

Comment:
Can we move this function to the top of the module, and add a bunch of unittests?

Reply:
and `os.path` and `pathlib` are utilities to work with os-related paths. i.e. it uses back-slashes on windows and not forward-slashes as required on the web.